internal/core/v1beta1/options.go

```diff
@@ -0,0 +1,20 @@
+package v1beta1
```

Comment:
These seem like generic API options.  Do they need to be versioned?

Reply:
I guess these options do not need to be versioned.

But I'm not sure how to extract them out without creating a circular dependency.
For example if change this package to `core`, then the references to `DistributionService` in this file would need to be changed to `v1beta1.DistributionService`. Which is a dependency on `package v1beta1`.
Then in distribution_service.go would need to refer to the `Option` as `core.Option`, which creates a circular dependency.

Currently the `WithConfig` and `WithCloudfrontClient` options are generic at the [server](https://github.com/kanopy-platform/cdnvalidator/blob/DEVOPS-3053/v1beta1_DistributionService/internal/server/options.go) level, and translated to v1beta1 options there. The same level as when the [v1beta1.New()](https://github.com/kanopy-platform/cdnvalidator/blob/DEVOPS-3053/v1beta1_DistributionService/internal/server/server.go#L45) API is constructed.

Reply:
These make not make sense as options then, since there is no default that can be provided. Config and CloudFront Client are required therefore they make the most sense as being constructor args on Server.

1.  server.go New(conf, cloudFront, opts)
    - server.go creates the v1beta1.DistbrutionService
2. v1beta1/api.go - New(router,  v1beta1.DistributionService).  This function no longer needs to return an error.